crypto/txt_db/txt_db.c

```diff
@@ -21,9 +21,8 @@ TXT_DB *TXT_DB_read(BIO *in, int num)
 {
     TXT_DB *ret = NULL;
     int esc = 0;
-    int i, add, n;
-    int size = BUFSIZE;
-    int offset = 0;
+    int n;
+    size_t i, add, size = BUFSIZE, offset = 0, unum = num;
```

Comment:
A check for `num < 0` is necessary. The original loop `for (i = 0; i < num; i++) {` would not execute, but now, the loop could execute excessively overwriting memory and potentially crash.

Reply:
You're right. Good catch.